pkg/asset/tls/certkey.go

```diff
@@ -0,0 +1,153 @@
+package tls
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// CertKey contains the private key and the cert that's
+// signed by the parent CA.
+type CertKey struct {
+	// assetStock is a reference to the store so that
+	// we can get the InstallConfig instance.
+	assetStock *asset.Stock
+
+	// Common fields.
+	Subject      pkix.Name
+	Validity     time.Duration
+	KeyUsages    x509.KeyUsage
+	ExtKeyUsages []x509.ExtKeyUsage
+	KeyFileName  string
+	CertFileName string
+	ParentCA     asset.Asset
+
+	IsCA         bool
+	IsRoot       bool
+	AppendParent bool // Whether append the parent CA in the cert.
```

Comment:
I don't see you setting this `false` anywhere:

```console
$ git grep AppendParent origin/pr/145
origin/pr/145:pkg/asset/tls/certkey.go: AppendParent bool // Whether append the parent CA in the cert.
origin/pr/145:pkg/asset/tls/certkey.go:         key, crt, err = generateCert(caKey, caCert, cfg, c.AppendParent)
origin/pr/145:pkg/asset/tls/variables.go:               AppendParent: true,
origin/pr/145:pkg/asset/tls/variables.go:               AppendParent:   true,
origin/pr/145:pkg/asset/tls/variables.go:               AppendParent:   true,
origin/pr/145:pkg/asset/tls/variables.go:               AppendParent: true,
```

Can we drop it?

Reply:
> For CAs I didn't set it...

Ah, that explains my `grep`.  Can we just use `!IsCA` then, or do we need to toggle this independently sometimes?

Reply:
only server certs need to append the signing CA; client certs don't need append.

Also, IsCA vs IsRoot ?

https://github.com/openshift/library-go/blob/master/pkg/crypto/crypto.go

Reply:
> only server certs need to append the signing CA; client certs don't need append.

Can we distinguish that using `ExtKeyUsageClientAuth`?

> Also, IsCA vs IsRoot ?

Won't root just be "`IsCA` set, but `ParentCA` unset"?

Reply:
The IsCA boolean is required by the `x509.Certificate` type.

> Can we distinguish that using ExtKeyUsageClientAuth?
Seems we can't, the `apiserver-proxy` cert still has `ExtKeyUsageClientAuth` but it doesn't have the `AppendParent`


BTW, I split the root into a separate type as it shares less code with others.